zebra-chain/src/block/difficulty.rs

```diff
@@ -93,6 +93,45 @@ impl fmt::Debug for ExpandedDifficulty {
     }
 }
 
+/// A 128-bit unsigned "Work" value.
+///
+/// Used to calculate the total work for each chain of blocks.
+///
+/// Details:
+///
+/// The relative value of `Work` is consensus-critical, because it is used to
+/// choose the best chain. But its precise value and bit pattern are not
+/// consensus-critical.
+///
+/// We calculate work values according to the Zcash specification, but store
+/// them as u128, rather than the implied u256. We don't expect the total chain
+/// work to ever exceed 2^128. The current total chain work for Zcash is 2^58,
+/// and Bitcoin adds around 2^91 work per year. (Each extra bit represents twice
+/// as much work.)
+#[derive(Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
+pub struct Work(u128);
+
+impl fmt::Debug for Work {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let log2_work = f64::from_str(&self.0.to_string()).ok().map(f64::log2);
+
+        // There isn't a standard way to represent alternate formats for the
+        // same value.
+        let mut f = f.debug_tuple("Work");
+        // Use hex, because expanded difficulty is in hex.
+        f.field(&format_args!("{:#x}", self.0))
+            // Use decimal, to compare with zcashd
+            .field(&format_args!("{}", self.0));
+
+        if let Some(log2_work) = log2_work {
+            // Use log2, to compare with zcashd
+            f.field(&format_args!("{:.5}", log2_work));
+        }
+
+        f.finish()
```

Comment:
```suggestion
        f.debug_tuple("Work")
        // Use hex, because expanded difficulty is in hex.
            .field(&format_args!("{:#x}", self.0))
            // Use decimal, to compare with zcashd
            .field(&format_args!("{}", self.0))
            // Use log2, to compare with zcashd
            .field(&format_args!("{:.5}", log2_work))
            .finish()
```